edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudConnection.cs

```diff
@@ -34,7 +35,8 @@ class CloudConnection : ICloudConnection
             ICloudListener cloudListener,
             TimeSpan idleTimeout,
             bool closeOnIdleTimeout,
-            TimeSpan operationTimeout)
+            TimeSpan operationTimeout,
+            string productInfo)
```

Comment:
Can this string ever be null?

Reply:
If not we should probably assert it, or if it can be convert to option?

Reply:
No, it can be empty string, but never null.